ir/instr.cpp

```diff
@@ -1443,13 +1537,24 @@ StateValue FpTernaryOp::toSMT(State &s) const {
     };
     break;
   }
-
-  auto scalar = [&](const StateValue &a, const StateValue &b,
-                    const StateValue &c, const Type &ty) {
+  
+  scalar = [&](const StateValue &a, const StateValue &b,
+               const StateValue &c, const Type &ty) {
     return fm_poison(s, a.value, a.non_poison, b.value, b.non_poison,
                      c.value, c.non_poison, fn, ty, fmath, rm, false);
   };
 
+  if (config::is_uf_float()) {
+    scalar = [&](const StateValue &a, const StateValue &b,
+                      const StateValue &c, const Type &ty) -> StateValue {
+      s.doesApproximation("uf_float", true);
```

Comment:
these tags are not very helpful like this. The 2nd argument should be the created UF.